subprojects/core/src/main/java/org/gradle/api/internal/tasks/SnapshotTaskInputsOperationDetails.java

```diff
@@ -28,20 +30,19 @@
  *
  * This operation is executed only when the build cache is enabled.
  *
- * This class is intentionally internal and consumed by the build scan plugin.
- *
  * @since 4.0
  */
+@UsedByScanPlugin
 public final class SnapshotTaskInputsOperationDetails implements BuildOperationDetails<SnapshotTaskInputsOperationDetails.Result> {
 
-    private final String taskPath;
+    private final Task task;
 
-    public SnapshotTaskInputsOperationDetails(String taskPath) {
-        this.taskPath = taskPath;
+    public SnapshotTaskInputsOperationDetails(Task task) {
+        this.task = task;
     }
 
-    public String getTaskPath() {
-        return taskPath;
+    public Task getTask() {
```

Comment:
I just talked with @adammurdoch about this: You could use the identity hashCode instead (which will be unique since we hold on to both tasks for the full duration of the build). It would be really great if we could avoid leaking the task object. We definitely can't commit to the `Task` being the contract long term.